### ISSUE TITLE
Profile - separate first and last name.

### ISSUE THREAD

**Issue by thiagotalma** [dektrium/yii2-user#332]
Dmitry, usually social networks have the record of the person as first name, middle name, last name.

In the profile table, there is only one "name" field.
It would be nice that this field was separated. At least two (first and last).

What do you think?

**Comment by jsobers1331** [comment]
This would be useful I think... Could we add it as a feature request @dmeroff ?

**Comment by dmeroff** [comment]
Yeah, it is good suggestion to get profile attributes from social network. I see this in such way: we introduce new property to the clients classes named $profileAttributeMap, which will look like:

```php
$profileAttributeMap = [
// social network attribute => profile attribute
'lastname' => 'name',
'site' => 'website',
// etc.
];
```
So developers will be able to add new attributes to profile and then map networks attributes to them on registration step.

Unfortunately it can not be done right now, because profile creation is hardcoded at the moment and before implementing this I should close #236

**Comment by SamMousa** [comment]
How about creating a schema.org Person interface and mapping it to those properties instead?
http://schema.org/Person

That way you have a "standard"  to adhere to.

**Comment by nikhiltri** [comment]
I like having a single name field, as not all names will fit nicely into a first-middle-last breakdown. See: https://www.w3.org/International/questions/qa-personal-names.

**Comment by dmeroff** [comment]
Closed in favor of #755
